File: calculadora-java/src/main/java/org/example/controller/CalculatorController.java
package org.example.controller;

import org.example.model.CalculatorModel;
import org.example.model.dto.RequestDTO;
import org.example.model.operations.IOperation;
import org.example.view.CalculatorView;

import java.lang.reflect.InvocationTargetException;
import java.util.List;

public class CalculatorController {
    private CalculatorView view;
    private CalculatorModel model;

    public CalculatorController(CalculatorView view, CalculatorModel model) {
        this.view = view;
        this.model = model;
    }

    public RequestDTO showMenuAndGetInput() {
        List<Class<? extends IOperation>> classList = model.getClassesNames();
        model.sortClassList(classList);
        view.dynamicMenu(classList);

        return view.collectInput();
    }

    public IOperation getOperation(int option) {
        List<Class<? extends IOperation>> classList = model.getClassesNames();

        try {
            for (Class<? extends IOperation> clazz : classList) {
                IOperation instance = (IOperation) clazz.getDeclaredConstructor().newInstance();
                int id = instance.getId();
                if (option == id) {
                    return instance;
                };
            }
        } catch (NoSuchMethodException | InstantiationException |
                 IllegalAccessException | InvocationTargetException e) {
            System.out.print("erro");
        }

        return null;
    }

    public Float calculate(RequestDTO dto) {
        int option = (int) dto.getOption();

        IOperation operation = getOperation(option);

        return operation.operate(dto.getNumber1(), dto.getNumber2());
    }


/*
    public Float calculate() {
        int optionInt = (int) requestDTO.getOption();
        float number1 = requestDTO.getNumber1();
        float number2 = requestDTO.getNumber2();

        try {
            return switch (optionInt) {
                case 1 -> model.somar(number1, number2);
                case 2 -> model.subtrair(number1, number2);
                case 3 -> model.multiplicar(number1, number2);
                case 4 -> model.dividir(number1, number2);
                default -> throw new IllegalStateException();
            };
        } catch (ArithmeticException e) {
            System.out.print("não pode divisão por zeroooo >:(");
        }

        return null;
    }
*/
}

File: calculadora-csharp/src/controller/CalculatorController.cs
using System;
using System.Collections.Generic;
using calculadora_csharp.model;
using calculadora_csharp.model.dto;
using calculadora_csharp.model.operations;
using calculadora_csharp.view;

namespace calculadora_csharp.controller;

public class CalculatorController(CalculatorView view, CalculatorModel model)
{
    public CalculatorView View = view;
    public CalculatorModel Model = model;

    public RequestDto ShowMenuAndGetInput()
    {
        List<Type> classList = model.GetClassesNames();
        model.SortClassList(classList);
        view.DynamicMenu(classList);

        return view.CollectInput();
    }

    public IOperation GetOperation(int option)
    {
        List<Type> classList = model.GetClassesNames();

        try
        {
            foreach (var clazz in classList)
            {
                var instance = Activator.CreateInstance(clazz) as IOperation;
                int id = instance?.Id ?? -1;

                if (option == id)
                {
                    return instance;
                }
            }
        }
        catch (Exception e)
        {
            Console.WriteLine(e);
            throw;
        }

        return null;
    }

    public float Calculate(RequestDto dto)
    {
        var option = (int) dto.Option;
        
        if (option == 0) { return 0; }
        
        var operation = GetOperation(option);

        if (operation == null)
        {
            throw new Exception("operação nula.");
        }
        else
        {
            return operation.Operate(dto.Number1, dto.Number2);
        }
    }
}